### DISCUSSION TITLE
`brew install` should not update every other package

### DISCUSSION THREAD

**Discussion by callamd** [Homebrew/discussions#4310] [Everyday usage]
### Output of `brew config`

```shell
irrelevant
```

### Output of `brew doctor`

```shell
also
```

### Description of issue

When running an install command for a new package brew should not spend 20 minutes updating all of my other packages before even attempting to do what I've asked it to.

I know my intentions much better than brew. Brew should first install my package so I can carry on with my life, then warn me my packages could be updated. It took brew 15 minutes of slowly updating everything and then failing before even getting to my installation. It took me 2 seconds to install the package with snap.

It's not very unixy at all, It's a package manager, not a windows installation. **brew should not do this**.

**Comment by SMillerDev** [comment]
But then your package would be broken because it depends on those updated versions.

https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-or-brew-install-formula-also-upgrade-a-bunch-of-other-stuff

**Reply by callamd** [reply]
`scrcpy` doesn't work with python3.10.1 so needs python 3.10.10? Sounds unlikely.

**Reply by SMillerDev** [reply]
In the meantime the default python changed so it's likely refering to python by a different path in the newer version. Long story short, most of the time it's needed.
